lab-miketreat/test/server-test.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const server = require('../server.js');
+const expect = require('chai').expect;
+const net = require('net');
+
+
+describe('server', ()=>{
+  before(done => {
+    server.listen(3000);
+    done();
+  });
+
+  after(done => {
+    server.close();
+    done();
+
+  });
+
+  // describe('Name Change', () => {
+  //   it('should change the username of the current user!', done => {
+  //     let updatedNickName = 'Mike';
+  //     let client = net.connect({port: 3000}, ()=> {
```

Comment:
so you'll want to do the .write command first, then outside of this net.connect close it, then you would do the .on('data'), also you'll want to end() the client before calling done. Done signals that the it block is finished.